From 1f1051260492855843b2032cb212b61626f3bcea Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 13 Feb 2024 19:58:35 -0800
Subject: [PATCH 1/2] hwdb: add resolution override for Pinebook Pro touchpad

The Pinebook Pro touchpad returns a resolution data that is 2 times of
the real value, which makes libinput think the touchpad is only 1/4 the
real size.

Add a resolution override value for it, to allow libinput to calculate
the distance moved on it correctly.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 hwdb.d/60-evdev.hwdb | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hwdb.d/60-evdev.hwdb b/hwdb.d/60-evdev.hwdb
index 9a90420b5f..7a0c756b5e 100644
--- a/hwdb.d/60-evdev.hwdb
+++ b/hwdb.d/60-evdev.hwdb
@@ -487,6 +487,17 @@ evdev:input:b0003v256Cp0064*
  EVDEV_ABS_00=::200
  EVDEV_ABS_01=::200
 
+###########################################################
+# Pine64
+###########################################################
+
+# Pinebook Pro
+evdev:input:b0003v258Ap001E*
+ EVDEV_ABS_00=::15
+ EVDEV_ABS_01=::15
+ EVDEV_ABS_35=::15
+ EVDEV_ABS_36=::15
+
 #########################################
 # Lenovo
 #########################################
-- 
2.43.0

